src/components/Search/SearchRouter/SearchRouter.tsx

```diff
@@ -45,21 +75,46 @@ function SearchRouter() {
         }
     }, SEARCH_DEBOUNCE_DELAY);
 
-    const onSearchSubmit = useCallback(() => {
+    const closeAndClearRouter = useCallback(() => {
         closeSearchRouter();
-
-        const query = SearchUtils.buildSearchQueryString(currentQuery);
-        Navigation.navigate(ROUTES.SEARCH_CENTRAL_PANE.getRoute({query}));
         clearUserQuery();
-    }, [currentQuery, closeSearchRouter]);
+    }, [closeSearchRouter]);
+
+    const onSearchSubmit = useCallback(
+        (query: SearchQueryJSON | undefined) => {
+            if (!query) {
+                return;
+            }
+            closeSearchRouter();
+            const queryString = SearchUtils.buildSearchQueryString(query);
+            Navigation.navigate(ROUTES.SEARCH_CENTRAL_PANE.getRoute({query: queryString}));
+            clearUserQuery();
+        },
+        [closeSearchRouter],
+    );
 
+<<<<<<< HEAD
+    useKeyboardShortcut(
+        CONST.KEYBOARD_SHORTCUTS.ENTER,
+        () => {
+            onSearchSubmit(currentQuery);
+        },
+        {
+            captureOnInputs: true,
+            shouldBubble: false,
+        },
+    );
+
+=======
+>>>>>>> main
     useKeyboardShortcut(CONST.KEYBOARD_SHORTCUTS.ESCAPE, () => {
         closeSearchRouter();
         clearUserQuery();
     });
 
     const modalType = isSmallScreenWidth ? CONST.MODAL.MODAL_TYPE.CENTERED : CONST.MODAL.MODAL_TYPE.POPOVER;
-    const isFullWidth = isSmallScreenWidth;
+    const isFullScreen = isSmallScreenWidth;
```

Comment:
I think it will be confusing for people to see:
`isFullScreen = isSmallScreen` - we have the context and understand why this happens, but for other devs I think it might be confusing 😅 

Since you stopped passing `isFullWidth` down to `SearchRouterInput` then I would completely drop this variable, and instead just do:
`const modalWidth = isSmallScreenWidth ? ...`
and 
`const modalListHeight = isSmallScreenWidth ? ... : styles.mh85vh`

then just use the consts